zerver/tests/test_converters.py

```diff
@@ -0,0 +1,226 @@
+from decimal import Decimal
+from enum import Enum
+from typing import Dict, List, Literal, Optional, Sequence, Type, TypedDict
+
+import orjson
+from attr import define
+from cattrs import StructureHandlerNotFoundError, transform_error
+
+from zerver.lib.converters import (
+    MAX_INT_SIZE,
+    Color,
+    NonNegativeInt,
+    URLStr,
+    forbid_extra_keys,
+    format_exception,
+    json_converter,
+    register_hook_for_converters,
+    zulip_converter,
+)
+from zerver.lib.test_classes import ZulipTestCase
+
+
+class ConvertersTestCase(ZulipTestCase):
+    class _NoOutput:
+        def __repr__(self) -> str:
+            return "<NoOutput>"
+
+    NoOutput = _NoOutput()
+
+    def _assert_converter_results(
+        self,
+        input: object,
+        target_type: Type[object],
+        expected_messages: Sequence[str] = [],
+        use_json: bool = True,
+    ) -> object:
```

Comment:
I really try to avoid boolean arguments in test functions.  Would it be easier to just have two helper functions here?  I know you'd duplicate some of the exception handling code, but is the exception handling code really necessary for somebody to debug the test?

Reply:
This is used internally by the `verify_formatter_output` helper, and the exception handling here is essential to capture the validation errors raised by `cattrs`.